File: Second year/MPP/Java_MPP_Bun/src/main/java/com/example/java_mpp_bun/domain/Entity.java
package com.example.java_mpp_bun.domain;

import java.io.Serializable;

public class Entity <ID> implements Serializable {
    private ID id;

    /**
     * getter id
     * @return id
     */
    public ID getID(){
        return id;
    }

    /**
     * setter id
     * @param id_new the new value of id
     */
    public void setID(ID id_new){
        id=id_new;
    }
}

File: Second year/ISS/ISS_Restaurant_Interface/ISS_Restaurant_Interface/domain/Entity.cs
using System;

namespace ISS_Restaurant_Interface.domain
{
    [Serializable]
    public abstract class Entity<ID>
    {
        public ID id { get; set;}

        public Entity(ID newId)
        {
            id = newId;
        }

        public abstract string to_string();
    }
}